### DISCUSSION TITLE
Running 'git log ' command on actions fails

### DISCUSSION THREAD

**Discussion by Narimm** [community/community#36723] [Actions] [answered]
I have created a github action that uses ant to build a package.
the act task looks like 

`<target name="changelog.generate">
        <property name="git_arg" value='log --pretty=format:"* %s (%an)" ${version.prior}..${version.current}'/>
        <exec executable="git" failonerror="true" outputproperty="git.changelog" logerror="true" resolveexecutable="true">
            <arg line='${git_arg}'/>
        </exec>
        <echo message="${git.changelog}"/>
        <delete file="${build.dir}\${plugin.slug}\CHANGELOG" failonerror="true"/>
        <concat destfile="${build.dir}\${plugin.slug}\CHANGELOG"  fixlastline="true">
            <header trimleading="yes">${TODAY} - v${version.prior} .. v${version.current}
                ${git.changelog}
            </header>
            <fileset dir="." includes="CHANGELOG" />
        </concat>
    </target>`

version.prior is a git tag 
version.current is `HEAD`

Locally when I run this on ubuntu I get :
`changelog.generate:
     [exec] Current OS is Linux
     [exec] Output redirected to property: git.changelog
     [exec] Executing 'git' with arguments:
     [exec] 'log'
     [exec] '--pretty=format:* %s (%an)'
     [exec] '5.6.2..HEAD'
     [exec]
     [exec] The ' characters around the executable and arguments are
     [exec] not part of the command.
     [echo] * ...(list of changes)
`
However on github actions it fails with the [message](https://github.com/Narimm/software-license-manager/actions/runs/3285787938/jobs/5413217659#step:4:30)

`changelog.generate:
     [exec] Current OS is Linux
     [exec] Output redirected to property: git.changelog
     [exec] Executing 'git' with arguments:
     [exec] 'log'
     [exec] '--pretty=format:* %s (%an)'
     [exec] '5.6.2..HEAD'
     [exec] 
     [exec] The ' characters around the executable and arguments are
     [exec] not part of the command.
     [exec] fatal: ambiguous argument '5.6.2..HEAD': unknown revision or path not in the working tree.
     [exec] Use '--' to separate paths from revisions, like this:
     [exec] 'git <command> [<revision>...] -- [<file>...]'
`

Anyone know why git log seemingly behaves differently on github actions

**Comment by Narimm** [comment, marked as answer]
Git hub actions by default does not checkout tags 
Adding an action to checkout remote tags solves the error/

**Comment by ossanna16** [comment]
Hey there @Narimm and thank you for asking a good question! I'm glad you found our community 🙂
